ddtrace/context.py

```diff
@@ -37,17 +36,31 @@ def __init__(
         self.span_id = span_id
 
         self._lock = threading.RLock()
-        # TODO[v1.0]: we need to keep a reference back to the span to maintain
-        # backwards compatibility when using context as a parent.
-        # eg:
-        #    ctx = span.context
-        #    tracer.start_span(child_of=ctx)
-        self._span = None  # type: Optional[Span]
         self._meta = {}  # type: _MetaDictType
         self._metrics = {}  # type: _MetricDictType
         self.dd_origin = dd_origin
         self.sampling_priority = sampling_priority
 
+    def __eq__(self, other):
+        return (
+            self.span_id == other.span_id
+            and self.trace_id == other.trace_id
+            and self.sampling_priority == other.sampling_priority
+            and self.dd_origin == other.dd_origin
+            and self._meta == other._meta
+            and self._metrics == other._metrics
+        )
+
+    def _with_span(self, span):
```

Comment:
This should probably be "public", given that's used outside of the module?

Reply:
Correct - I just don't want to expose this as an API to users yet